siliconcompiler/asic.py

```
@@ -670,67 +670,243 @@ def set_asic_var(self,
+            seen = set()
+            while value.startswith("$"):
+                if value in seen or value not in sdc_vars:
+                    return None
+                seen.add(value)
+                value = sdc_vars[value]
+            return value
+
+        for command in commands:
+            # A create_clock inside a comment or a quoted string is not a clock
+            # definition; one after a semicolon is.
+            for command in re.findall(r"^\s*create_clock\s(.*)", command):
```

Ruff (0.16.1)

[warning] Loop control variable `command` overrides iterable it iterates

(B020)